### DISCUSSION TITLE
Can we replace garden with containerd?

### DISCUSSION THREAD

**Discussion by kkarthik23** [concourse/concourse#6634] [Help & Support] [answered]
Can we replace garden with containerd ? if so how do we do this?

Also if someone could please explain  how  garden works behind  in the background  it would be great ?

**Comment by fauna-chase** [comment, marked as answer]
I know that in the v7.0.0 release of concourse you can switch to containerd, instructions pulled from the release notes, for more details go here: https://github.com/concourse/concourse/releases/tag/v7.0.0



The formerly-experimental containerd runtime is now GA and is considered ready for production use 🔗

    We will be changing the default container runtime from Guardian to containerd in coming releases, but we encourage using the containerd runtime ASAP
    To enable the containerd runtime, set --runtime ($CONCOURSE_RUNTIME) to containerd on the concourse worker command
    You will also need to convert any --garden-* ($CONCOURSE_GARDEN_*) flags to their containerd counterparts:



idk about previous versions

**Comment by taylorsilva** [comment]
There are a few layers here as it sounds like you noticed.

Each worker starts a garden server. You can think of this as a web server providing a REST API.

Behind the garden server is a specific container runtime: guardian, containerd, houdini

So what I think you really meant to ask was: how can I replace **guardian** with containerd. @fauna-chase answer's that question.

**Reply by kkarthik23** [reply]
thank you both
